src/modelbuilder.nit

```diff
@@ -38,14 +38,17 @@ redef class ToolContext
 	do
 		assert not mmodules.is_empty
 		var mainmodule
-		if mmodules.length == 1 then
+		# We need a main module, so we build it by importing all modules
+		mainmodule = new MModule(modelbuilder.model, null, mmodules.first.name + "-m", new Location(mmodules.first.location.file, 0, 0, 0, 0))
+		mainmodule.is_fictive = true
+		mainmodule.first_real_mmodule = mmodules.first.first_real_mmodule
+		mainmodule.set_imported_mmodules(mmodules)
+		modelbuilder.apply_conditional_importations(mainmodule)
+		if mainmodule.in_importation.direct_greaters.length == 1 and mainmodule.in_importation.direct_greaters.first == mmodules.first then
+			# Drop the fictive module if not needed
 			mainmodule = mmodules.first
 		else
-			# We need a main module, so we build it by importing all modules
-			mainmodule = new MModule(modelbuilder.model, null, mmodules.first.name + "-m", new Location(mmodules.first.location.file, 0, 0, 0, 0))
-			mainmodule.is_fictive = true
-			mainmodule.set_imported_mmodules(mmodules)
-			modelbuilder.apply_conditional_importations(mainmodule)
+			# Or else run phases on it
 			modelbuilder.run_phases
```

Comment:
It's just that if we plan to drop it, I don't the the point to do analysis on them.